File: src/main/java/com/github/thorben/webnd/domain/character/SkillTypeRepository.java
package com.github.thorben.webnd.domain.character;

import com.github.thorben.webnd.domain.games.GameType;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

@Repository
public interface SkillTypeRepository extends CrudRepository<SkillType, Integer> {

	List<SkillType> findAllByGameType(GameType gameType);

	Optional<SkillType> findByName(String name);

}

File: src/main/kotlin/com/github/thorben/webnd/domain/character/SkillTypeRepository.kt
package com.github.thorben.webnd.domain.character

import com.github.thorben.webnd.domain.games.GameType
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface SkillTypeRepository : CrudRepository<SkillType, Int> {
	fun findAllByGameType(gameType: GameType): List<SkillType>
	fun findByName(name: String): Optional<SkillType>
}